usl_pipeline/study_area_uploader/study_area_uploader/transformers/study_area_transformers.py

```diff
@@ -178,6 +201,15 @@ def prepare_and_upload_study_area_files(
                     mask_value_feature_property=soil_class_mask_feature_property,
                 )
             )
+            if len(input_non_green_area_soil_classes) > 0:
+                soil_classes_polygons = [
+                    _update_mask_if_needed(
```

Comment:
Is the idea here to collapse the different non-green-area soil classes into a single non-green-area value? Would it also work to continue passing the set `input_non_green_area_soil_classes` around, or does it need to be a single value?

Reply:
Not for the need to also do the same nearest neighbor procedure for every chunk of the full study area (NYC) in cloud_functions. If we could pass around user-provided parameters to cloud_functions, then I wouldn't substitute non-green soil classes by some default value.